lib/chef/provider/aws_subnet.rb

```diff
@@ -147,9 +167,13 @@ def update_network_acl(subnet)
     if new_resource.network_acl
       network_acl_id =
         AWSResource.lookup_options({ network_acl: new_resource.network_acl }, resource: new_resource)[:network_acl]
-      if subnet.network_acl.id != network_acl_id
+      # Below snippet gives the entry of network_acl who is associated with current subnet by matching its subnet_id
+      network_acl_association = subnet.client.describe_network_acls(filters: [{name: "vpc-id", values: [subnet.vpc.id]}, {name: "association.subnet-id", values: [subnet.id]}]).network_acls.first.associations
+      current_network_acl_association = network_acl_association.find {|r| r.subnet_id == subnet.id}
```

Comment:
Are we guaranteed to find a match?  Could associations be empty?  If so, we may want to guard against current_network_acl_association being `nil` at like 174.